src/mount-support.c

```diff
@@ -392,3 +392,23 @@ static char * __attribute__ ((used))
 	*offsetp = offset;
 	return (offset < fulllen) ? &path[offset] : NULL;
 }
+
+/*
+ * Check that @subdir is a subdir of @dir.
+ * @len is the length of @dir (to avoid having to recalculate it).
+**/
+static bool __attribute__ ((used))
+    is_subdir(const char *subdir, const char *dir, size_t len)
```

Comment:
This function comes straight from upstream LXD codebase. I agree on the size being awkward to pass. The original code doesn't really take any advantage of this. I will change the method to compute both sizes by itself.